src/hotspot/share/runtime/cpuTimeCounters.hpp

```diff
@@ -51,6 +53,8 @@ class CPUTimeGroups : public AllStatic {
 };
 
 class CPUTimeCounters: public CHeapObj<mtServiceability> {
+  friend class CPUTimeScope;
```

Comment:
I wonder if we avoid this. The public APIs should be enough.

Reply:
Oh, now I see what you mean. Thanks for pointing that out :)